core/src/com/pottda/game/controller/ControllerHookup.java

```diff
@@ -0,0 +1,91 @@
+package com.pottda.game.controller;
+
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.pottda.game.model.*;
+import com.pottda.game.model.Character;
+import com.pottda.game.view.ActorView;
+
+import javax.vecmath.Vector2f;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A class which implements NewModelListener
+ * <p>
+ * On new model, it creates a new controller and sets its view and model
+ */
+public class ControllerHookup implements NewModelListener {
+
+    /**
+     * The stage onto which the ViewActors are put
+     */
+    private final Stage stage;
+
+    public ControllerHookup(Stage stage) {
+        this.stage = stage;
+    }
+
+    private List<NewControllerListener> listenerList = new ArrayList<NewControllerListener>();
+
+    public void addListener(NewControllerListener ncl) {
+        listenerList.add(ncl);
+    }
+
+    @Override
+    public void onNewModel(ModelActor m) {
+
+        ActorView view = null;
+
+        // Determine what kind of controller
+        AbstractController controller = null;
+        if (m instanceof Projectile) {
+            view = new ActorView(m.sprite);
+            controller = new ProjectileController(m, view);
+        } else if (m instanceof Character) {
+            view = new ActorView(m.sprite);
+            if (m.team == Character.PLAYER_TEAM) {
+                controller = createInputController(m, view);
+
+            } else if (m.team == Character.ENEMY_TEAM) {
+                controller = createController(m, view);
+            }
+        } else if (m instanceof Obstacle) {
+            view = new ActorView(m.sprite, ((Vector2f) ((Obstacle) m).size));
+            controller = new ObstacleController(m, view);
+        }
+        if(view != null) {
```

Comment:
We should probably throw exceptions instead of null-checking: if we get null, something is very wrong